cont3xt/integration.js

```diff
@@ -623,43 +635,46 @@ class Integration {
       sent: 0,
       total: 0, // runIntegrationsList will fix
       resultCount: 0, // sum of _cont3xt.count from results
-      queriedSet: new Set()
+      queriedSet: new Set(),
+      finished: false,
+      finishWrite () { // end data write and create audit log when finished
+        res.write(JSON.stringify({ purpose: 'finish', resultCount: this.resultCount }));
+        res.end(']\n');
+
+        Audit.create({
+          userId: req.user.userId,
+          indicator: queries.join(', '),
+          iType: isBulk ? 'bulk' : indicators[0].itype,
+          tags: req.body.tags ?? [],
+          viewId: req.body.viewId,
+          issuedAt,
+          took: Date.now() - issuedAt,
+          resultCount: this.resultCount
+        }).catch((err) => {
+          console.log('ERROR - creating audit log.', err);
+        });
+      }
     };
     res.write('[\n');
-    res.write(JSON.stringify({ purpose: 'init', indicator, sent: shared.sent, total: integrations.length, text: 'more to follow' }));
+    res.write(JSON.stringify({ purpose: 'init', indicator: indicators[0], sent: shared.sent, total: Integration.#integrations[indicators[0].itype].length, text: 'more to follow' }));
```

Comment:
indicator from 'init' is used to set the initial active indicator in the UI. The order should stay the same as the queries we do here since each runIntegrationsList immediately sends a purpose:'link' which puts it into the order.

Reply:
I could go either way. If you sent indicators in init would that remove "link" purpose items that have no parents?

Reply:
Yea, we could get rid of the purpose:'link's with parentIndicator:undefined if we send indicators

Reply:
im good either way :)